### ISSUE TITLE
[Feature] Implement Modify responses

### ISSUE THREAD

**Issue by easyrepro** [microsoft/playwright-java#794]
Implement Modify responses as available for Node.js

https://playwright.dev/docs/network#modify-responses

**Comment by yury-s** [comment]
> Tried to ran below code with intent update response body and status only after request is completed but before it rendered on UI. The updated response was never served with below code. If I remove `route.resume()` then requests were not triggering so I have to keep `route.resume()`

You cannot call `fulfill` after `resume`, only one of the methods continue/fulfill/abort can be called. If you want to intercept response you should use [APIRequestContext](https://playwright.dev/java/docs/api/class-apirequestcontext) to get original response and then call fulfill with modified data.

`fulfill` is lacks `response` parameter compared to nodejs api, I'll add that but it should not be a blocker for your scenario as yuo can manually get headers, status and body from the [APIResponse](https://playwright.dev/java/docs/api/class-apiresponse).

**Comment by easyrepro** [comment]
Thanks for looking into the issue.

Adding `response `parameter to `fulfill` will greatly helps instead of manually setting headers, status and body from the [APIResponse](https://playwright.dev/java/docs/api/class-apiresponse)

I want to intercept for only one API service, tried with below code. Getting [APIRequestContext](https://playwright.dev/java/docs/api/class-apirequestcontext) from `page.context().request()` instead of directly converting `route.request()` of type [Request](https://playwright.dev/java/docs/api/class-request) seems a little more work. 

Any alternatives?

Ex:

`

       page.route("**/*", route -> {
            APIRequestContext apiRequestContext = page.context().request();
            APIResponse apiResponse = apiRequestContext.fetch(route.request());
            route.fulfill(new Route.FulfillOptions()
                    .setStatus(apiResponse.status())
                    .setBodyBytes(apiResponse.body())
                    .setHeaders(apiResponse.headers())
            );
        });
`

**Comment by easyrepro** [comment]
Thanks for the reply.

Using `page.context().request()` to get [APIRequestContext](https://playwright.dev/java/docs/api/class-apirequestcontext) inside `route` closure seems backward.

Any way to auto wrap `route.request()` of type [Request](https://playwright.dev/java/docs/api/class-request) to [APIRequestContext](https://playwright.dev/java/docs/api/class-apirequestcontext)

**Comment by yury-s** [comment]
No, `apiRequestContext.fetch` that you mentioned is the only way.
